php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -1469,22 +1473,15 @@ class NDB_BVL_Instrument extends NDB_Page
             return 'Complete';
         }
 
-        $db = \Database::singleton();
-
-        $query = "SELECT " . join(',', $this->_requiredElements)
-            . " FROM $this->table WHERE CommentID=:CID";
-
-        $dataFields = $db->pselectRow(
-            $query,
-            array('CID' => $this->getCommentID())
-        );
-        foreach ($dataFields as $field) {
+        $allData = $this->loadInstanceData($this);
+        foreach ($this->_requiredElements as $field) {
             // this shouldn't be just empty() b/c 0 is a valid
             // value for some required fields
-            if (is_null($field) || $field === "") {
+            if (is_null($allData[$field]) || $allData[$field] === "") {
```

Comment:
i think there might be someone wrong here. `Data_entry_completion_status` gets set to Incomplete even though it should be complete / was complete before

Reply:
@ridz1208 did you have any issues with scoring when setting jsonData to true? when i enabled using the flag table, the value for `Data_entry_completion_status` column for that instrument always gets saved as "incomplete" and the code doesn't enter the instrument score function as a result